station/tests/test_quality_ratings.py

```diff
@@ -0,0 +1,88 @@
+import unittest
+
+import numpy as np
+
+import quality_ratings
+
+
+class TestQualityRatings(unittest.TestCase):
+    def test_list_names(self):
+        names = quality_ratings.get_rate_names()
+        self.assertTrue("analog" in names)
+        self.assertTrue("digital" in names)
+
+    def test_get_rate_by_name(self):
+        self.assertIsNotNone(quality_ratings.get_rate_by_name("analog"))
+        self.assertIsNotNone(quality_ratings.get_rate_by_name("digital"))
+
+    def test_analog_rating_on_gaussian_noise_small_sigma(self):
+        img = np.random.normal(scale=1, size=(1000, 1000))
+        rate = quality_ratings.get_rate_by_name("analog")
+        rating = rate(img)
+        self.assertAlmostEquals(1.0, rating, 2)
+
+    def test_analog_rating_on_gaussian_noise_big_sigma(self):
+        img = np.random.normal(scale=20, size=(1000, 1000))
+        rate = quality_ratings.get_rate_by_name("analog")
+        rating = rate(img)
+        self.assertAlmostEquals(0.0, rating, 2)
+
+    def test_analog_rating_on_gaussian_noise_medium_sigma(self):
+        img = np.random.normal(scale=12.7, size=(1000, 1000))
+        rate = quality_ratings.get_rate_by_name("analog")
+        rating = rate(img)
+        self.assertAlmostEquals(0.5, rating, 1)
+
+    def test_analog_rating_on_blank(self):
+        img = np.zeros((1000, 1000))
+        rate = quality_ratings.get_rate_by_name("analog")
+        rating = rate(img)
+        self.assertAlmostEquals(1, rating, 2)
+
+    def test_digital_rating_on_black(self):
+        img = np.zeros((1000, 1000))
+        rate = quality_ratings.get_rate_by_name("digital")
+        rating = rate(img)
+        self.assertAlmostEquals(0, rating, 3)
+
+    def test_digital_rating_on_white(self):
+        img = np.ones((1000, 1000))
+        rate = quality_ratings.get_rate_by_name("digital")
+        rating = rate(img)
+        self.assertAlmostEquals(1, rating, 3)
+
+    def test_digital_rating_on_half_zeros(self):
+        z = np.ones((1000, 500))
+        o = np.zeros((1000, 500))
+        img = np.hstack((z, o))
+        rate = quality_ratings.get_rate_by_name("digital")
+        rating = rate(img)
+        self.assertAlmostEquals(0.5, rating, 3)
+
+    def test_digital_rating_on_quater_zeros(self):
+        ones = np.ones((1000, 750))
+        zeros = np.zeros((1000, 250))
+        img = np.hstack((zeros, ones))
+        rate = quality_ratings.get_rate_by_name("digital")
+        rating = rate(img)
+        self.assertAlmostEquals(0.75, rating, 3)
+
+    def test_digital_rating_on_3d_black(self):
+        img = np.zeros((1000, 1000, 3))
```

Comment:
So this creates 3 planes, each being 1000x1000. Does this have any specific practical application, maybe representing 3 (RGB?) channels of a 1000x1000 image? That's ok if it doesn't and this is just an artificial test case.

Reply:
It is data order compatible with OpenCV layout - one layer with 3D pixels (1000, 1000, 3). If you used before PIL/Pillow then you may be used to 3 layers layout (3, 1000, 1000).